nncf/experimental/common/pruning/operations.py

```diff
@@ -0,0 +1,637 @@
+        The dimension block encoding pruning of a single channel from S is no longer valid.
+        This function creates new blocks that encodes how many channels from S is pruned if we prune along a new
+        dimension from [A,B,C,D].
+        It forms new constraints by the following rules:
+            | size  | offset          |
+            |-------|-----------------|
+            | 1     | D % S           |
+            | D     | C*D % S         |
+            | C*D   | B*C*D % S       |
+            | B*C*D | A*B*C*D % S = 0 |
+
+        :param block: dimension block to split
+        :param list_output_channels: list of output channels (A,B,C,D)
+        :return:
+        """
+        if len(list_output_channels) == 1:
+            raise RuntimeError
+
+        dot_product = reduce((lambda x, y: x * y), list_output_channels)
+
+        current_size = dot_product
+        new_blocks = []
+        divided_shapes = filter(lambda x: x != 1, list_output_channels)
+        for divided_shape in divided_shapes:
+            offset = int(current_size % dot_product)
+            current_size /= divided_shape
+            new_block = deepcopy(block)
+            new_block.size = int(current_size)
+            new_block.offset = int(offset)
+            new_blocks.append(new_block)
+        return new_blocks
+
+    @classmethod
+    def _split_group(cls,
+                     group: PropagationGroup,
+                     input_channels: int,
+                     list_output_channels: List[int]) -> List[PropagationGroup]:
+        """
+        Splits a group of dimension blocks into new child groups by a pruning dimension.
+        Internally calls "_split_block" to split each block from the given group. Please refer to the
+        description of the method for more details about splitting blocks.
+        Group can have multiple blocks. E.g. s1 and s2. Let's say s1 is splitted into (a1,b1), s2 - into (a2,b2).
+        The method will return 2 childs groups: (a1,a2) and (b1,b2).
+
+        :param input_channels: splitted input channels (S).
+        :param list_output_channels: list of output channels (A,B,C,D).
+        :param group: a group of dimension blocks for splitting.
+        :return: list of new groups which resultes from the split.
+        """
+        new_blocks: List[List[PropagationBlock]] = []
+        child_groups: List[PropagationGroup] = []
+
+        for block in group.get_blocks():
+            dot_product = reduce((lambda x, y: x * y), list_output_channels)
+            assert dot_product == input_channels
+            new_blocks.append(cls._split_block(block, list_output_channels))
+
+        # forms [(a1,a2), (b1,b2)] from [(a1,b1), (a2,b2)]
+        for block_groups in zip(*new_blocks):
+            child_groups.append(PropagationGroup(blocks=list(block_groups)))
+        for child_group in child_groups:
+            group.add_child(child_group)
```

Comment:
```suggestion
        for block_groups in zip(*new_blocks):
            group.add_child(PropagationGroup(blocks=list(block_groups)))
```

Reply:
But we will have 1 loop instead of 2 loops

Reply:
ok, changed